plugin/trino-phoenix/src/test/java/io/trino/plugin/phoenix/TestPhoenixTypeMapping.java

```diff
@@ -256,110 +279,129 @@ public void testDecimalUnspecifiedPrecision()
-        checkIsGap(someZone, dateOfLocalTimeChangeForwardAtMidnightInSomeZone.atStartOfDay());
-        LocalDate dateOfLocalTimeChangeBackwardAtMidnightInSomeZone = LocalDate.of(1983, 10, 1);
-        checkIsDoubled(someZone, dateOfLocalTimeChangeBackwardAtMidnightInSomeZone.atStartOfDay().minusMinutes(1));
-
-        DataTypeTest trinoTestCases = dateTests(
-                dateOfLocalTimeChangeForwardAtMidnightInJvmZone,
-                dateOfLocalTimeChangeForwardAtMidnightInSomeZone,
-                dateOfLocalTimeChangeBackwardAtMidnightInSomeZone,
-                dateDataType());
-
-        DataTypeTest phoenixTestCases = dateTests(
-                dateOfLocalTimeChangeForwardAtMidnightInJvmZone,
-                dateOfLocalTimeChangeForwardAtMidnightInSomeZone,
-                dateOfLocalTimeChangeBackwardAtMidnightInSomeZone,
-                phoenixDateDataType())
-                .addRoundTrip(primaryKey(), 1);
-
-        for (String timeZoneId : ImmutableList.of(UTC_KEY.getId(), jvmZone.getId(), someZone.getId())) {
-            Session session = Session.builder(getSession())
-                    .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(timeZoneId))
-                    .build();
-            trinoTestCases.execute(getQueryRunner(), session, trinoCreateAsSelect(session, "test_date"));
-            trinoTestCases.execute(getQueryRunner(), session, trinoCreateAsSelect(getSession(), "test_date"));
-            trinoTestCases.execute(getQueryRunner(), session, trinoCreateAndInsert(session, "test_date"));
-            phoenixTestCases.execute(getQueryRunner(), session, phoenixCreateAndInsert("tpch.test_date"));
-        }
+        SqlDataTypeTest.create()
+                .addRoundTrip("date", "TO_DATE('1952-04-04', 'yyyy-MM-dd', 'local')", DATE, "DATE '1952-04-04'") // before epoch
+                .addRoundTrip("date", "TO_DATE('1970-01-01', 'yyyy-MM-dd', 'local')", DATE, "DATE '1970-01-01'")
+                .addRoundTrip("date", "TO_DATE('1970-02-03', 'yyyy-MM-dd', 'local')", DATE, "DATE '1970-02-03'")
+                .addRoundTrip("date", "TO_DATE('2017-07-01', 'yyyy-MM-dd', 'local')", DATE, "DATE '2017-07-01'") // summer on northern hemisphere (possible DST)
+                .addRoundTrip("date", "TO_DATE('2017-01-01', 'yyyy-MM-dd', 'local')", DATE, "DATE '2017-01-01'") // winter on northern hemisphere (possible DST on southern hemisphere)
+                .addRoundTrip("date", "TO_DATE('1983-04-01', 'yyyy-MM-dd', 'local')", DATE, "DATE '1983-04-01'")
+                .addRoundTrip("date", "TO_DATE('1983-10-01', 'yyyy-MM-dd', 'local')", DATE, "DATE '1983-10-01'")
+                .addRoundTrip("date", "NULL", DATE, "CAST(NULL AS DATE)")
+                .addRoundTrip("integer primary key", "1", INTEGER, "1")
+                .execute(getQueryRunner(), session, phoenixCreateAndInsert("tpch.test_date"));
     }
 
     @Test
     public void testArray()
```

Comment:
I notice we aren't consistently testing arrays with both 1 and more than 1 elements.

e.g. bigint we only test with single element arrays. My justification is that some systems (e.g. ElasticSearch) treat single element arrays differently from others.

Also arrays containing NULLs look like another good test which is already covered in `testArrayNulls` but not for all data types.